tensorlayer/models/vgg19.py

```diff
@@ -0,0 +1,392 @@
+                net, n_filter=256, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv3_1'
+            ),
+            lambda net: Conv2d(
+                net, n_filter=256, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv3_2'
+            ),
+            lambda net: Conv2d(
+                net, n_filter=256, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv3_3'
+            ),
+            lambda net: Conv2d(
+                net, n_filter=256, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv3_4'
+            ),
+            lambda net: MaxPool2d(net, filter_size=(2, 2), strides=(2, 2), padding='SAME', name='pool3'),
+
+            # conv4
+            lambda net: Conv2d(
+                net, n_filter=512, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv4_1'
+            ),
+            lambda net: Conv2d(
+                net, n_filter=512, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv4_2'
+            ),
+            lambda net: Conv2d(
+                net, n_filter=512, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv4_3'
+            ),
+            lambda net: Conv2d(
+                net, n_filter=512, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv4_4'
+            ),
+            lambda net: MaxPool2d(net, filter_size=(2, 2), strides=(2, 2), padding='SAME', name='pool4'),
+
+            # conv5
+            lambda net: Conv2d(
+                net, n_filter=512, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv5_1'
+            ),
+            lambda net: Conv2d(
+                net, n_filter=512, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv5_2'
+            ),
+            lambda net: Conv2d(
+                net, n_filter=512, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv5_3'
+            ),
+            lambda net: Conv2d(
+                net, n_filter=512, filter_size=(3, 3), strides=(1, 1), act=tf.nn.relu, padding='SAME', name='conv5_4'
+            ),
+            lambda net: MaxPool2d(net, filter_size=(2, 2), strides=(2, 2), padding='SAME', name='pool5'),
+            lambda net: FlattenLayer(net, name='flatten'),
+            lambda net: DenseLayer(net, n_units=4096, act=tf.nn.relu, name='fc1_relu'),
+            lambda net: DenseLayer(net, n_units=4096, act=tf.nn.relu, name='fc2_relu'),
+            lambda net: DenseLayer(net, n_units=1000, act=tf.identity, name='fc3_relu'),
+        ]
+        net = net_in
+        for l in layers:
+            net = l(net)
+            # if end_with in net.name:
+            if net.name.endswith(end_with):
+                return net
+
+        raise Exception("unknown layer name (end_with): {}".format(end_with))
+
+    def restore_params(self, sess):
```

Comment:
we should make it download automatically. @lgarithm @luomai  any idea?

Reply:
One simple solution is that we train all the models and store theses models in the libraryof the TL

Reply:
I got a "This site can’t be reached" error when I tried to download the npy of VGG'19 from the website given. 

How large is the npy file? Can we host it in our pre-trained model repo?

Reply:
Yes, it is about 570MB for this npy file. The public repo maximum  limit for the file is 100 MB. However, I do not know what the limit is in the private repo.

Reply:
The original download link is for browser download only.
I've uploaded it to tensorlayer/pretrained-models, you can now use this download link in the program:

```
https://media.githubusercontent.com/media/tensorlayer/pretrained-models/master/models/vgg19.npy
```